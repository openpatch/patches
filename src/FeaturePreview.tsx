/** @jsx jsx */
import { ReactNode } from "react";
import { jsx, css } from "@emotion/core";
import { Grid } from "./Grid";
import { Heading } from "./Heading";
import { Text } from "./Text";
import { Video } from "./Video";
import { Center } from "./Center";

export type FeaturePreviewProps = {
  src: string;
  title: string;
  description: string;
  maxWidth?: number;
  reverse?: boolean;
};

export const FeaturePreview = ({
  src,
  title,
  description,
  maxWidth = 500,
}: FeaturePreviewProps) => {
  const descriptionBox = (
    <Center>
      <Heading as="h3">{title}</Heading>
      <Text textAlign="center">{description}</Text>
    </Center>
  );

  const videoBox = (
    <Center position="relative">
      <Video
        src={src}
        boxShadow="large"
        borderRadius="standard"
        loop
        autoPlay
        style={{
          width: "100%",
          maxWidth,
        }}
      ></Video>
    </Center>
  );
  return (
    <Grid gridTemplateColumns={["1fr", "1fr auto"]} gridGap="large">
      {descriptionBox}
      {videoBox}
    </Grid>
  );
};
