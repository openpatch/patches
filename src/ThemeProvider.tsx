import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
