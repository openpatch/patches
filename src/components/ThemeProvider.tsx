import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
