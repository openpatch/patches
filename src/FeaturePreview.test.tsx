import React from "react";
import { render } from "@testing-library/react";
import { FeaturePreview, FeaturePreviewProps } from "./FeaturePreview";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FeaturePreviewProps) =>
  render(
    <ThemeProvider>
      <FeaturePreview
        title="Interactive Tasks"
        description="Highlighting, Puzzles and Coding to name a few. Interactive tasks are a great way to make use of the digital format. Each tasks can automatically be evaluated."
        src="https://f002.backblazeb2.com/file/openpatch-static/interactive.mp4"
        {...props}
      />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
