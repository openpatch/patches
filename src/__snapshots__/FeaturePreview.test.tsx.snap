// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 32px;
}

@media screen and (min-width: 768px) {
  .emotion-0 {
    grid-template-columns: 1fr auto;
  }
}

.emotion-2 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
}

.emotion-6 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  text-align: center;
}

.emotion-8 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-10 {
  border-radius: 8px;
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0 emotion-1"
      display="grid"
    >
      <div
        class="emotion-2 emotion-1"
        display="flex"
      >
        <h3
          class="emotion-4 emotion-1"
        >
          Interactive Tasks
        </h3>
        <p
          class="emotion-6 emotion-1"
        >
          Highlighting, Puzzles and Coding to name a few. Interactive tasks are a great way to make use of the digital format. Each tasks can automatically be evaluated.
        </p>
      </div>
      <div
        class="emotion-8 emotion-1"
        display="flex"
      >
        <video
          autoplay=""
          class="emotion-10 emotion-11"
          loop=""
          style="width: 100%; max-width: 500px;"
        >
          <source
            src="https://f002.backblazeb2.com/file/openpatch-static/interactive.mp4"
          />
        </video>
      </div>
    </div>
  </div>
</div>
`;
